src/main/java/de/tum/in/www1/artemis/service/AttachmentUnitService.java

```diff
@@ -66,6 +74,36 @@ public AttachmentUnit createAttachmentUnit(AttachmentUnit attachmentUnit, Attach
         return savedAttachmentUnit;
     }
 
+    /**
+     * Creates new attachment units for the given lecture.
+     * @param lectureUnitSplitDTOs The split information
+     * @param lecture The lecture linked to the attachmentUnits
+     * @param file The file (lecture slide) to be split
+     * @return The created attachment units
+     */
+    public List<AttachmentUnit> createAttachmentUnits(List<LectureUnitSplitDTO> lectureUnitSplitDTOs, Lecture lecture, MultipartFile file) throws IOException {
+        List<Long> ids = new ArrayList<>();
+        List<LectureUnitDTO> lectureUnitsDTO = lectureUnitProcessingService.splitUnits(lectureUnitSplitDTOs, file);
+        lectureUnitsDTO.forEach(lectureUnit -> {
+            lectureUnit.attachmentUnit().setLecture(null);
+            AttachmentUnit savedAttachmentUnit = attachmentUnitRepository.saveAndFlush(lectureUnit.attachmentUnit());
+            ids.add(savedAttachmentUnit.getId());
+            lectureUnit.attachmentUnit().setLecture(lecture);
+            lecture.addLectureUnit(savedAttachmentUnit);
+            lectureRepository.save(lecture);
+
+            lectureUnit.attachment().setVersion(0);
```

Comment:
You are calling `attachment.setVersion(1);` in the service, why are you going back to 0 here?

Reply:
You are right, it should not be set in unit processing service but only in attachment unit service. 

Currently version update is not working because version was not set correctly. Now i removed the version increase from handleFile method which was called in couple places, to updateAttachment method.